packages/dds/tree/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -986,9 +986,9 @@ function setMarkAdjacentCells(mark: Mark, adjacentCells: IdRange[]): void {
 function shouldReceiveLineage(
 	cellRevision: RevisionTag | undefined,
 	detachRevision: RevisionTag,
-	metadata: RevisionMetadataSource,
+	metadata: RebaseRevisionMetadata,
 ): boolean {
-	if (cellRevision === undefined) {
+	if (cellRevision === undefined || cellRevision === metadata.getRevisionToRebase()) {
```

Comment:
Was this a bug before?

Reply:
Arguably. Previously there was an assumption that a changeset being rebased always had implicit (undefined) revision, which was weird but worked fine in practice.